app/controllers/divisions_controller.rb

```diff
@@ -31,8 +31,14 @@ def index
         @year = params[:date]
       elsif params[:date] =~/^\d{4}-\d{2}$/
         @month = params[:date]
-      else
-        @date = params[:date]
+      elsif params[:date]
```

Comment:
I totally agree with it. But I am not familiar with the code and I like baby steps. That's why I just did a simple fix. I tried to add a test, but there was an error that I couldn't figure out how to fix it. 

If we fix that, I can add more tests and learn exactly what this controller is doing. Then the refactor tasks gets much easier.